### PR TITLE
Test history refactoring and improvements

### PR DESCRIPTION
**Pull request by mdealer**
See https://github.com/mdealer/junit-plugin/releases/tag/junit-1.63.2-refactor-test-history

TL;DR:
- Major performance and usability improvements to the test history file storage, history charts and test failure age computation.

TODOs:
- [x] Make sure you are opening from a **topic/feature/bugfix branch** (right side) and not your main branch!
- [x] Ensure that the pull request title represents the desired changelog entry
- [x] Please describe what you did
- [ ] Link to relevant issues in GitHub or Jira
- [x] Link to relevant pull requests, esp. upstream and downstream changes
- [ ] Provide new tests
- [x] Fix some existing tests 
- [ ] sync with latest master

Some guidance on what is the best way to debug the automated tests without resorting to logging (that would take ages) would be helpful. Based on continuous live/production usage on our side, it works.

Live 'tested' with multibranch and plain pipeline jobs with JUnit and TestNG test reports. Newer versions of xUnit seem to have issues due to this being based on the older junit-plugin 1.63

### CODE REVIEW

**Comment by mdealer** [comment]
> > Some guidance on what is the best way to debug the automated tests without resorting to logging (that would take ages) would be helpful. Based on continuous live/production usage on our side, it works.
> 
> you should just be able to open them in a Java IDE and put breakpoints in and run the test in debug mode. Nothing special required in this project

Can VS Code do this? I tried putting some breakpoints and pressing F5 but it does nothing. What IDE do you use?

**Comment by timja** [comment]
I (and most other Java developers) use IntelliJ.

Eclipse should work but I haven't touched it in ~8 years so couldn't say.

VS Code should be able but I've never tried.

**Comment by mdealer** [comment]
Thanks, after trying around a bunch of times, I settled on IntelliJ.

**Comment by mdealer** [comment]
Updated via https://github.com/jenkinsci/junit-plugin/pull/625
